### ISSUE TITLE
Ghidra crashes Ubuntu 18.04

### ISSUE THREAD

**Issue by filipkilibarda** [NationalSecurityAgency/ghidra#653]
This question is now more of a notice to others experiencing similar issues.

`Ubuntu 18.04.2`
`Gnome-Shell 3.28.3`

The problem is that Ghidra causes my `gnome-shell` session to crash with a segfault *sometimes* when I focus the Ghidra window. When `gnome-shell` restarts, I'm back at the login screen with all previously running applications having been killed.

```
Jun 03 12:47:58 xps kernel: gnome-shell[28318]: segfault at 28 ip 00007f7ba48fd280 sp 00007ffdae7b1028 error 4 in libmutter-2.so.0.0.0[7f7ba4879000+158000]
```

[Possible related issue](https://gitlab.gnome.org/GNOME/mutter/issues/308) in the `mutter` repo in the GNOME project. I posted there to see if it could be resolved on the `gnome-shell` side. 

JetBrains IDE had a similar issue and it seems that it was resolved on the IDE side by using a different window type. So fixing it on the Ghidra side might be possible.

Will probably just end up running Ghidra in a VM window so that all my other applications aren't killed whenever this happens.

UPDATE: The issue is fixed in GNOME Shell 3.32.1. Solution for me was to install Ubuntu 19.04 in a VM and run Ghidra out of there.

**Comment by ryanmkurtz** [comment]
Have you modified any of the JVM arguments in launch.properties?

**Comment by Ayrx** [comment]
FWIW I have no stability issues running Ghidra on Ubuntu 18.04 with the default JVM configurations.

**Comment by filipkilibarda** [comment]
> Have you modified any of the JVM arguments in launch.properties?

I haven't. I just downloaded the latest release (9.0.4) and I'm running it without any modifications. I also had this problem with the initial release (9.0).

I figured out that this crash produces a core dump from `/usr/bin/Xwayland`, so seems like this is a bug in the X server. For some reason it only happens *sometimes* when I focus the Ghidra application, i.e., I switch to its workspace.

Last few lines of the core dump as viewed in `gdb`.

```
Core was generated by `/usr/bin/Xwayland :0 -rootless -terminate -accessx -core -listen 4 -listen 5 -d'.
Program terminated with signal SIGABRT, Aborted.
#0  0x00007f488d3d3e97 in raise () from /lib/x86_64-linux-gnu/libc.so.6
```

Seems that Xwayland is just killing itself for some reason. (sending a SIGABRT to itself)

**Comment by ryanmkurtz** [comment]
You could try flipping `VMARGS=-Dsun.java2d.xrender=false` to `true` in support/launch.properties just as a place to start.

**Comment by filipkilibarda** [comment]
> You could try flipping `VMARGS=-Dsun.java2d.xrender=false` to `true` in support/launch.properties just as a place to start.

Doesn't seem to fix it. Log output from Xwayland below. Its complaining about a mundane window resizing issue prior to the crash. I'll try looking at the javascript code to see where it's crashing exactly.

```
Jun 03 11:55:48 xps org.gnome.Shell.desktop[28318]: glamor: EGL version 1.4 (DRI2):
Jun 03 12:02:44 xps org.gnome.Shell.desktop[28318]: Window manager warning: last_user_time (56801049) is greater than comparison timestamp (56801046).  This most likely represents a buggy client sending inaccurate timestamps in messages such as _NET_ACTIVE_WINDOW.  Trying to work around...
Jun 03 12:02:44 xps org.gnome.Shell.desktop[28318]: Window manager warning: W2 (fkilibarda) appears to be one of the offending windows with a timestamp of 56801049.  Working around...
Jun 03 12:17:47 xps org.gnome.Shell.desktop[28318]: Window manager warning: Window 0x2400007 (Ghidra) sets an MWM hint indicating it isn't resizable, but sets min size 1 x 1 and max size 2147483647 x 2147483647; this doesn't make much sense.
Jun 03 12:17:47 xps org.gnome.Shell.desktop[28318]: Window manager warning: Window 0x2400020 (win0) sets an MWM hint indicating it isn't resizable, but sets min size 1 x 1 and max size 2147483647 x 2147483647; this doesn't make much sense.
Jun 03 12:17:47 xps org.gnome.Shell.desktop[28318]: Window manager warning: Invalid WM_TRANSIENT_FOR window 0x0 specified for 0x2400020 (win0).
Jun 03 12:47:48 xps org.gnome.Shell.desktop[28318]: Window manager warning: Window 0x2400516 (win1) sets an MWM hint indicating it isn't resizable, but sets min size 1 x 1 and max size 2147483647 x 2147483647; this doesn't make much sense.
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: == Stack trace for context 0x5603569df340 ==
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: #0 0x560356d70a20 i   resource:///org/gnome/shell/ui/windowManager.js:1796 (0x7f7b59d23bc0 @ 113)
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: #1 0x7ffdae7b1fe0 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7f7b7c2b5de0 @ 71)
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: #2 0x560356d70998 i   resource:///org/gnome/shell/ui/tweener.js:112 (0x7f7b7c2c8e68 @ 37)
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: #3 0x7ffdae7b3a50 b   resource:///org/gnome/gjs/modules/tweener/tweener.js:208 (0x7f7b7c2d2b38 @ 54)
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: #4 0x7ffdae7b4310 b   resource:///org/gnome/gjs/modules/tweener/tweener.js:337 (0x7f7b7c2d2bc0 @ 1626)
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: #5 0x7ffdae7b43c0 b   resource:///org/gnome/gjs/modules/tweener/tweener.js:350 (0x7f7b7c2d2c48 @ 100)
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: #6 0x560356d70920 i   resource:///org/gnome/gjs/modules/tweener/tweener.js:365 (0x7f7b7c2d2cd0 @ 10)
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: #7 0x7ffdae7b5740 b   resource:///org/gnome/gjs/modules/signals.js:128 (0x7f7b7c2d2230 @ 386)
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: #8 0x560356d70890 i   resource:///org/gnome/shell/ui/tweener.js:244 (0x7f7b7c2cf808 @ 159)
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: #9 0x7ffdae7b6390 I   resource:///org/gnome/gjs/modules/_legacy.js:82 (0x7f7b7c2b5de0 @ 71)
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: #10 0x560356d70810 i   resource:///org/gnome/shell/ui/tweener.js:219 (0x7f7b7c2cf780 @ 15)
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: (EE)
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: Fatal server error:
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: (EE) failed to read Wayland events: Broken pipe
Jun 03 12:47:58 xps org.gnome.Shell.desktop[28318]: (EE)
Jun 03 12:47:58 xps gnome-session[28287]: gnome-session-binary[28287]: WARNING: Application 'org.gnome.Shell.desktop' killed by signal 11
Jun 03 12:47:58 xps gnome-session-binary[28287]: Unrecoverable failure in required component org.gnome.Shell.desktop
Jun 03 12:47:58 xps gnome-session-binary[28287]: WARNING: Application 'org.gnome.Shell.desktop' killed by signal 11
```

**Comment by filipkilibarda** [comment]
Looks like it's a segfault in `gnome-shell` 

```
Jun 03 12:47:58 xps kernel: gnome-shell[28318]: segfault at 28 ip 00007f7ba48fd280 sp 00007ffdae7b1028 error 4 in libmutter-2.so.0.0.0[7f7ba4879000+158000]
```

:(

**Comment by filipkilibarda** [comment]
[Possible related issue](https://gitlab.gnome.org/GNOME/mutter/issues/308), will post back here if I get an update from the Gnome Shell people.

**Comment by ryanmkurtz** [comment]
I'm going to close this for now since it seems like it is something that has to be fixed in Gnome and there is nothing the Ghidra team can do about it.  We can reopen it if there are any new developments.

**Comment by filipkilibarda** [comment]
Ok. Issue is fixed in GNOME Shell 3.32.1 which comes with Ubuntu 19.04. But Ubuntu 18.04 LTS is still on GNOME Shell 3.28.3, so wouldn't be surprised if others experience this
